### PR TITLE
added figure for book2/5.1

### PR DESCRIPTION
**Pull request by shobro**
## Description

Latexified, jaxified and added figure for `book2/5.1`
### Figure Number

`fig book2/5.1`
### Figures

|Before PR|After PR|
|---|---|
|Screenshot from 2022-05-30 18-09-31|Screenshot from 2022-06-06 01-21-45|
### Issue 
#55 

## Steps

- [x] Step1 Generated code for `fig book2/5.1`
- [x] Step2 Latexifed code
- [x] Step3 Jaxified code 

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

### CODE REVIEW

File: notebooks/book2/05/minimize_kl_divergence.ipynb [hunk 1]
```diff
@@ -0,0 +1,230 @@
+{
```
Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #23.        %pip install -qq sample_data
</pre><p>why <code>sample_data</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #24.        import seaborn
</pre><p>write same as line #23</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #30.    latexify(width_scale_factor=1.2, fig_height=1.5, font_size=9)
</pre><p>Please don't touch font_size until extremely needed, you have only one figure and lot of space, you don't need to decrease font size from 12 to 9</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    def kl_sampling(theta, q, samples=100000):  # loss function when KL(p:q)
</pre><p>use <code>params</code> instead of <code>theta</code>   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.        for i in range(n_itr):
</pre><p>You should use <code>jax.lax.scan()</code> now probably you know how to use that </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <p>Here you should add two plots for loss vs iterations.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    params_one = fit(params=params_one, optimizer=optimizer, loss_fun=kl_sampling, n_itr=n_itr)
</pre><p>Write <code>optimized_params_one</code> instead </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    plt.plot(x, q.prob(x), label="q")
</pre><p>I would suggest to use 3 different line style, for example <code>normal</code>, <code>dashed</code>,  <code>dash-dotted</code>. </p><p>Reason why I'm saying is, once Dr. Kevin mentioned that sometimes color blind people read this books and they won't able to distinguish these three lines, but different style can be easily recognized  </p><p>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #30.    latexify(width_scale_factor=1.2, fig_height=1.5, font_size=9)
</pre><p>set <code>fig_height = 1</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Comment by karm-patel** [comment]
Your legend labels are same , it should be min (p,q) and min (q,p)

**Comment by karm-patel** [comment]
And you can click on `resolve conversation` once comments are resolved

File: notebooks/book2/05/minimize_kl_divergence.ipynb [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
```
Comment (karm-patel) [inline notebooks/book2/05/minimize_kl_divergence.ipynb:1]: <p>For first plot, loss does not looks converged completely, I think you should increase number of ierations?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Comment by karm-patel** [comment]
Prof. @nipunbatra, would you like to give your comments?
